### ISSUE TITLE
It lints excluded paths in 6.16.2

### ISSUE THREAD

**Issue by ElieDeloumeau** [ansible/ansible-lint#3477]
##### Summary

ansible-lint is linting excluded paths since 6.16.2

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.16.2 using ansible-core:2.15.0 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

* Install collections in project/collections (awx.awx 19.2.2 in this case)
* Exclude paths in `.ansible-lint`
  * ```
    exclude_paths:
      - project/collections/
    ```
* Run `ansible-lint` and see it lint the collections

With ansible-lint 6.16.1 (normal)
```console
$ ansible-lint --force-color --offline
[...]
yaml[line-length]: Line too long (161 > 160 characters)
project/vars/awx/org_admin.yml:30
                Rule Violation Summary
 count tag               profile rule associated tags
     1 yaml[line-length] basic   formatting, yaml

Failed after min profile: 1 failure(s), 0 warning(s) on 14 files.
```

With ansible-lint 6.16.**2** (bug)
```console
$ ansible-lint --force-color --offline
[...]
risky-file-permissions: File permissions unset or incorrect. (warning)
project/collections/ansible_collections/awx/awx/tools/template_galaxy.yml:36 Task/Handler: Template the README.md source file (should be commited with your changes)

yaml[line-length]: Line too long (161 > 160 characters)
project/vars/awx/org_admin.yml:30

Read documentation for instructions on how to ignore specific rule violations.

                           Rule Violation Summary
 count tag                        profile    rule associated tags
     1 command-instead-of-module  basic      command-shell, idiom
     2 command-instead-of-shell   basic      command-shell, idiom
     2 deprecated-local-action    basic      deprecations
     1 key-order[task]            basic      formatting
     3 jinja[invalid]             basic      formatting
    18 jinja[spacing]             basic      formatting
   328 name[missing]              basic      idiom
     1 var-naming[no-reserved]    basic      idiom
    10 var-naming[no-role-prefix] basic      idiom
     7 yaml[line-length]          basic      formatting, yaml
     1 name[template]             moderate   idiom
     8 risky-file-permissions     safety     unpredictability (warning)
     2 no-changed-when            shared     command-shell, idempotency
   450 fqcn[action-core]          production formatting (warning)
    28 args[module]                          syntax, experimental (warning)

Failed after min profile: 376 failure(s), 486 warning(s) on 172 files.
```

##### Desired Behavior

Excluded paths should be excluded from linting

##### Actual Behavior

Excluded paths are linted

**Comment by chrisx8** [comment]
Same here on `6.16.2`, both `--exclude` and `exclude_paths` have no effect. Works as expected on 6.16.1 and earlier.

**Comment by OlivierCloudar** [comment]
I can confirm the same behaviour

**Comment by cayla** [comment]
I wonder if https://github.com/ansible/ansible-lint/pull/3468/files is related.  That was an obvious change in 6.16.2

PS - You can just thumbsup the original post if you are affected.  You don't have to say 'me too'.

**Comment by chrisx8** [comment]
Reverting #3468 (commit ab184059) solved the issue

**Comment by OlivierCloudar** [comment]
I am not using it in pre-commit, but similar to the first post:

- a `.ansible-lint` file with the `exclude_paths` directive containing a list of paths to ignore
- calling the executable directly with `ansible-lint` without any extra arguments

This works if I use v6.16.1 but stopped working since the release of v6.16.2

**Comment by Dual-0** [comment]
I exclude it via `.ansible-lint` config file.

```yaml
exclude_paths:
  - .cache/ # implicit unless exclude_paths is defined in config
  - .git/
  - roles/monitoring/files/alerts.d/
```
exec:
```bash
user@instance:~/ansible$ ansible-lint
WARNING  Listing 30 violation(s) that are fatal
yaml[trailing-spaces]: Trailing spaces
roles/monitoring/files/alerts.d/dl.yml:5

yaml[line-length]: Line too long (178 > 160 characters)
roles/monitoring/files/alerts.d/dl.yml:6
```

**Comment by ssbarnea** [comment]
Found at least what is causing it, a trailing `/` on folders. If you do not use a trailing `/` the excluded will work as indented.

Obviously that I don't plan to document the need to remove slashes, so I am looking to make a PR to address it.

**Comment by kpfleming** [comment]
The folder paths in my `.ansible-lint` file do not have trailing slashes, but they are still being included in the scan using 6.16.2.

```
%YAML 1.2
---
exclude_paths:
  - .cache
  - requirements.yml
  - roles/willshersystems.sshd
  - venv
  - collections
  - factcache
  - playbooks/test.yml

skip_list:
  - latest[git]
  - meta-no-info
  - name[casing]
  - no-changed-when
  - no-handler
  - package-latest
  - risky-shell-pipe
  - template-instead-of-copy
  - unnamed-task
  - var-naming[no-jinja]
  - var-naming[no-role-prefix]
  - var-spacing
  - yaml[line-length]

write_list:
  - all

kinds:
  - tasks: "**/imports/*.yml"
```

**Comment by ssbarnea** [comment]
If one of you can confirm if https://github.com/ansible/ansible-lint/pull/3487 is fixing the issue it would be great. I will wait for at least one confirmation before getting the fix in and making a new release.

**Comment by chrisx8** [comment]
I'm also still having the same issue with #3487. 

See below (there is no `.ansible-lint` file):

```shell
$ ansible-lint --version
ansible-lint 6.16.3.dev9 using ansible-core:2.15.0 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
You are using a pre-release version of ansible-lint.

$ ansible-lint --exclude roles/tailscale
WARNING  Listing 1 violation(s) that are fatal
fqcn[action-core]: Use FQCN for builtin module actions (lineinfile).
roles/tailscale/tasks/main.yml:16 Use `ansible.builtin.lineinfile` or `ansible.legacy.lineinfile` instead.

Read documentation for instructions on how to ignore specific rule violations.

                 Rule Violation Summary
 count tag               profile    rule associated tags
     1 fqcn[action-core] production formatting

Failed after shared profile, 4/5 star rating: 1 failure(s), 0 warning(s) on 135 files.
You are using a pre-release version of ansible-lint.
```

Note that a violation is found at `roles/tailscale/tasks/main.yml:16`, even if it should have been excluded.

**Comment by ssbarnea** [comment]
Please point me to some repository where I can reproduce the bug, posting console dumps is not always enough for reproducing a bug. Quite often some other settings are triggering the bug.

**Comment by osfrickler** [comment]
> Please point me to some repository where I can reproduce the bug, posting console dumps is not always enough for reproducing a bug. Quite often some other settings are triggering the bug.

https://github.com/osism/ansible-playbooks/ is where I first discovered this.

**Comment by ssbarnea** [comment]
I just pushed a rewrite of detection location that drops use of wcmatch, please test https://github.com/ansible/ansible-lint/pull/3487 and report the outcomes.

This impacts case where linter is called *without any positional arguments* in locations that are not git repos.

**Comment by chrisx8** [comment]
#3487 fixed the issue for me. Thanks @ssbarnea!

**Comment by hille721** [comment]
@ssbarnea : I can confirm that the fix will work for me as well
